packages/backend/src/remote/activitypub/resolver.ts

```diff
@@ -70,4 +89,52 @@ export default class Resolver {
 
 		return object;
 	}
+
+	private resolveLocal(url: string): Promise<IObject> {
+		const localOffset = config.url.length + 1; //includes slash after config.url
+
+		if (!url.startsWith(config.url + '/') || !url.slice(localOffset).includes('/')) {
+			throw new Error(`resolveLocal: unrecognized URL ${url}`);
+		}
+
+		const i = url.slice(localOffset).indexOf('/');
+		const type = url.slice(localOffset, localOffset + i);
+		const id = url.slice(localOffset + i + 1);
+
+		const activitySuffix = '/activity';
```

Comment:
Can't you share the URI separation routine with the following or use a regular expression in one shot?
https://github.com/misskey-dev/misskey/blob/fee2878b98253f2510ae2bc5c746c058a043ef82/packages/backend/src/remote/activitypub/db-resolver.ts#L124-L141

Reply:
Maybe it should be made `static` so I do not need to make an instance of `DbResolver` to use it.

Reply:
I wonder if `DbResolver` should have a general resolving function. If `Resolver` detects that a URL is local it could then delegate to `DbResolver`.